src/neynarAPI/NeynarAPIClient.ts

```diff
@@ -0,0 +1,679 @@
+import { silentLogger, Logger } from "../logger";
+import type { WithRequired } from "../utils";
+
+const BASE_PATH = "https://api.neynar.com/";
+
+export class NeynarAPIClient {
+  private readonly logger: Logger;
+
+  public readonly apis: {
+    user: UserApi;
+    v1Cast: CastApiV1;
+    verification: VerificationApi;
+    notifications: NotificationsApi;
+    reactions: ReactionsApi;
+    v1Follows: FollowsApiV1;
+    signer: SignerApi;
+    v2Cast: CastApiV2;
+    reaction: ReactionApi;
+    v2Follow: FollowApiV2;
+  };
+
+  /**
+   * Instantiates a NeynarV1APIClient
+   *
+   * Note: A Wallet must be provided if the API client is to mint new AuthTokens
+   */
+  constructor(
+    apiKey: string,
+    {
+      logger = silentLogger,
+      axiosInstance,
+    }: { logger?: Logger; axiosInstance?: AxiosInstance } = {}
+  ) {
+    this.logger = logger;
+
+    if (apiKey === undefined) {
+      throw new Error(
+        "Attempt to use an authenticated API method without first providing an api key"
+      );
+    }
+
+    if (axiosInstance === undefined) {
+      axiosInstance = axios.create();
+    }
+    axiosInstance.defaults.decompress = true;
+    axiosInstance.interceptors.response.use(
+      (response) => response,
+      (error) => {
+        if (NeynarAPIClient.isApiErrorResponse(error)) {
+          const apiErrors = error.response.data;
+          this.logger.warn(`API errors: ${JSON.stringify(apiErrors)}`);
+        }
+        throw error;
+      }
+    );
+
+    const v1Config: Configuration = new Configuration({
+      basePath: BASE_PATH + "v1",
+      apiKey: apiKey,
+    });
+    const v2Config: Configuration = new Configuration({
+      basePath: BASE_PATH + "v2",
+      apiKey: apiKey,
+    });
+    this.apis = {
+      v1Cast: new CastApiV1(v1Config, undefined, axiosInstance),
+      user: new UserApi(v1Config, undefined, axiosInstance),
+      verification: new VerificationApi(v1Config, undefined, axiosInstance),
+      notifications: new NotificationsApi(v1Config, undefined, axiosInstance),
+      reactions: new ReactionsApi(v1Config, undefined, axiosInstance),
+      v1Follows: new FollowsApiV1(v1Config, undefined, axiosInstance),
+      v2Cast: new CastApiV2(v2Config, undefined, axiosInstance),
+      signer: new SignerApi(v2Config, undefined, axiosInstance),
+      reaction: new ReactionApi(v2Config, undefined, axiosInstance),
+      v2Follow: new FollowApiV2(v2Config, undefined, axiosInstance),
+    };
+  }
+
+  /**
+   * Utility for parsing errors returned by the Merkle API server. Returns true
+   * if the given error is caused by an error response from the server, and
+   * narrows the type of `error` accordingly.
+   */
+  public static isApiErrorResponse(
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    error: any
+  ): error is WithRequired<AxiosError<ErrorRes>, "response"> {
+    if (!(error instanceof AxiosError)) return false;
+    return error.response?.data !== undefined;
```

Comment:
Curious for your thinking on why we check for existence of a `data` property on `response`

Reply:
The interceptor in the axios client set up in the constructor for the neynar client references the data field, so this makes sure that it exists

```
    axiosInstance.interceptors.response.use(
      (response) => response,
      (error) => {
        if (NeynarAPIClient.isApiErrorResponse(error)) {
          const apiErrors = error.response.data;
          this.logger.warn(`API errors: ${JSON.stringify(apiErrors)}`);
        }
        throw error;
      }
    );
```

Reply:
Sorry, I should have been more clear, I agree that this check is necessary to assert `error is WithRequired<AxiosError<ErrorRes>, "response>` but is it sufficient?

In the old MerkleAPIClient, for example, we also checked for the existence of an `errors` property in `error.response.data`